### ISSUE TITLE
Updates before cran submission

### ISSUE THREAD

**Issue by danagibbon** [dkoslicki/PressPurtCoreAlg#15]
### Summary of Changes
1. `expected_num_switch` all zeros:
* I'm getting a puzzling error in regards to this. From Python it's fine, if I have the output saved to disk (csv file) running from R it's fine, but when the output is saved directly to an R object it is rounded/cut off and outputs a matrix of all zeros. 
* I have an issue open with the Reticulate folks as this only seems to be an issue going from python to R via Reticulate.

2. 0 based (python) to 1 based (R)
* I've changed the names to be 1 based in `PreprocessMatrix` and `ComputeEntryWisePerturbationExpectation`

3. `get_distributions`
* I made a new function (for now so the objects saved to disk will still work) called `get_distributions_single`
* It takes the `matrix_interval`, `distribution_list`, `asymp_stab` interval and `points`
* If I misinterpreted that `points` is the number of `x_range` please let me know

4. Distributions object 
* I've added a new object to the output `ComputeEntryWisePerturbationExpectation` called `distributions_object` (I've kept the original for now)
* It is a list of lists of each coordinate pair and for each coordinate pair it contains the position, asymptotic stability interval,  and a dataframe of the distribution values and x range.
 * Here is a preview:
```
> lapply(Entrywise$distributions_object$`(1, 1)`, head)
$position
[1] 1 1

$interval
[1] -0.08298658  0.10901820

$distribution
      x_range dist_vals
1 -0.08490663  0.000000
2 -0.08412010  0.000000
3 -0.08333358  0.000000
4 -0.08254705  4.233417
5 -0.08176053  4.263202
6 -0.08097400  4.292908
```

5. NumSwitch
* List of matrix, one for each coordinate pair
* Has `num_switch_val`, `start`, `end`
* Still need to add step function
* Preview:
```
> Entrywise$num_switch_funcs_r
$`(1, 1)`
  num_switch_val       start       end
1              1 0.007114943 0.1090182

$`(1, 2)`
  num_switch_val start      end
1              1   -10 -0.03095

$`(2, 1)`
  num_switch_val    start end
1              1 0.003095  10

$`(2, 2)`
  num_switch_val       start         end
1              1  -0.3125131 -0.01305907
2              2 -10.0000000 -0.31251310
```

6. Separate scripts
* I split into `scripts.R` (main functions) and `helper_funcs.R`

7. Small Changes
* Added the original matrix to R object
* Split asymptotic_stability into asymptotic_stability_start & asymptotic_stability_end

### Still needed:
* Once new object format is finalized, update plotting function 
* Virtual environment 
* Implement `ComputeMultiEntryPerturbationExpectation.py`
* Convert objects saved to disk to 1 based (will write a function to turn into "R friendly" format) 
* Document how to update documentation

**Please note that the plotting function as not been updated yet so the output is not correct**

**Comment by danagibbon** [comment]
I've now added 
* `ComputeMultiEntryPerturbationExpectation` function
* `set_python_virtual` and virtual env to `py_depend`

**Comment by danagibbon** [comment]
New additions:
* Split functions up into `helper_funcs.R` and one `.R` for the 4 main functions
* Added `process_data` function to convert data saved to disk to R friendly format
* Fixed the expected num switch dataframe being all zeros issue
* Updated the man
* Added `ns_to_step` function
* Updated plotting function `GenerateEntryWiseFigures`

**Comment by danagibbon** [comment]
### Still needed:
* Restructure repo
* Update DESCRIPTION file with contact info
* Review man and see if any updates are needed
* Update documentation/vignette to include recent changes
* Submit to cran

**Comment by danagibbon** [comment]
still outstanding addressed in #16
